src/agni/server/receiver/UserReceiver.java

```diff
@@ -1,17 +1,55 @@
 package agni.server.receiver;
 
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Vector;
+
 public class UserReceiver implements MessageParser {
-    private UserListener[] userListeners;
+    private Vector <UserListener> userListeners = null;
 
     public UserReceiver() {
-
+        userListeners = new Vector<UserListener>();
     }
 
-    private void notifyUserRequest() {
+    private void notifyUserRequest(String ip, byte type, String message) {
```

Comment:
Probably in this PR we should be translating the `type` from type `byte` into some form of enum. Code for the enum would be:

```java
    public enum InfoType {
        JOIN_CHAT((byte)0x00),
        LEAVE_CHAT((byte)0x01),
        FRIENDS_LIST((byte)0x02);
        FRIEND_STATUS((byte)0x03);
        ADD_FRIEND((byte)0x04);
        LOGOUT((byte)0x05);

        private final byte bytes;
        private Status(byte bytes) {
            this.bytes = bytes;
        }

        public final byte bytes() {
            return bytes;
        }
    }
```